fuzzers/040-clk-hrow-config/top.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+import re
+import random
+random.seed(int(os.getenv("SEED"), 16))
+from prjxray import util
+from prjxray import verilog
+from prjxray.db import Database
+
+XY_RE = re.compile('^BUFHCE_X([0-9]+)Y([0-9]+)$')
+
+
+def gen_sites():
+    db = Database(util.get_db_root())
+    grid = db.grid()
+    db = Database(util.get_db_root())
+    grid = db.grid()
+    for tile_name in sorted(grid.tiles()):
+        loc = grid.loc_of_tilename(tile_name)
+        gridinfo = grid.gridinfo_at_loc(loc)
+        sites = []
+
+        xs = []
+        ys = []
+        for site, site_type in gridinfo.sites.items():
+            if site_type == 'BUFHCE':
+                m = re.match(XY_RE, site)
+                assert m, site
+                x = int(m.group(1))
+                y = int(m.group(2))
+                xs.append(x)
+                ys.append(y)
+
+                sites.append((site, x, y))
+
+        if sites:
+            yield tile_name, min(xs), min(ys), sorted(sites)
+
+
+def main():
+    print('''
+module top();
+    ''')
+
+    params_list = []
+    for tile_name, x_min, y_min, sites in gen_sites():
+
+        for site, x, y in sites:
+            params = {}
+            params['tile'] = tile_name
+            params['site'] = site
+            params['x'] = x - x_min
+            params['y'] = y - y_min
+            params['IN_USE'] = random.randint(0, 1)
+
+            if params['IN_USE']:
+                params['INIT_OUT'] = random.randint(0, 1)
+                params['CE_TYPE'] = verilog.quote(
+                    random.choice(('SYNC', 'ASYNC')))
+
+                print(
+                    '''
+    (* KEEP, DONT_TOUCH, LOC = "{site}" *)
+    BUFHCE #(
+        .INIT_OUT({INIT_OUT}),
+        .CE_TYPE({CE_TYPE})
+        ) buf_{site} ();
+                    '''.format(**params))
```

Comment:
I prefer templates like this not be defined inline, but recognize this is keeping with the pattern of existing fuzzers. Is there a reason to keep them inline?

Reply:
What is the advantage of moving them out of line?  That would appear to hurt readability?

Reply:
I guess it's preference. The multi-line template really breaks readability of functions for me with python's whitespace syntax. Having the templates together would let me see the parts of the eventual output. But I can see those that prefer have the template where it is used.